Mathlib/AlgebraicGeometry/Cover/Directed.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2025 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.AlgebraicGeometry.PullbackCarrier
+
+/-!
+# Directed covers
+
+A directed `P`-cover of a scheme `X` is a cover `𝒰` with an ordering
+on the indices and compatible transition maps `𝒰ᵢ ⟶ 𝒰ⱼ` for `i ≤ j` such that
+every `x : 𝒰ᵢ ×[X] 𝒰ⱼ` comes from some `𝒰ₖ` for a `k ≤ i` and `k ≤ j`.
+
+Gluing along directed covers is easier, because the intersections `𝒰ᵢ ×[X] 𝒰ⱼ` can
+be covered by a subcover of `𝒰`.
+
+Many natural covers are naturally directed, most importantly the cover of all affine
+opens of a scheme.
+-/
+
+universe u
+
+noncomputable section
+
+open CategoryTheory Limits
+
+namespace AlgebraicGeometry.Scheme
+
+variable {P : MorphismProperty Scheme.{u}} {X : Scheme.{u}}
+
+/-- A directed `P`-cover of a scheme `X` is a cover `𝒰` with an ordering
+on the indices and compatible transition maps `𝒰ᵢ ⟶ 𝒰ⱼ` for `i ≤ j` such that
+every `x : 𝒰ᵢ ×[X] 𝒰ⱼ` comes from some `𝒰ₖ` for a `k ≤ i` and `k ≤ j`. -/
+class Cover.Directed (𝒰 : X.Cover P) extends LE 𝒰.J where
+  /-- The transition map `𝒰ᵢ ⟶ 𝒰ⱼ` for `i ≤ j`. -/
+  trans {i j : 𝒰.J} (hij : i ≤ j) : 𝒰.obj i ⟶ 𝒰.obj j
```

Comment:
Do you not need functorial properties on `trans`? This is free for open covers because the `𝒰.map i`s are monomorphisms but for descent over general topology is it not necessary?